### PR TITLE
[SMALLFIX] Configuration cleanup

### PR DESCRIPTION
**Pull request by jsimsa**
- improves configuration initialization
- cleans up configuration validation
- adds configuration validation for test cluster

### CODE REVIEW

File: core/common/src/main/java/alluxio/Configuration.java [hunk 1]
```diff
@@ -493,58 +494,45 @@ private static void checkWorkerPorts() {
...
-
-  /**
-   * Checks that the configuration values are reasonable.
-   */
-  private static void checkConfigurationValues() {
-    checkWorkerPorts();
-    checkUserFileBufferBytes();
-    checkZkConfiguration();
+    Preconditions.checkState(
+        containsKey(PropertyKey.ZOOKEEPER_ADDRESS) == containsKey(PropertyKey.ZOOKEEPER_ADDRESS),
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/Configuration.java]: The equality check is not correct?

File: core/common/src/main/java/alluxio/Configuration.java [hunk 2]
```diff
@@ -493,58 +494,45 @@ private static void checkWorkerPorts() {
   }
 
   /**
-   * {@link PropertyKey#USER_FILE_BUFFER_BYTES} should not bigger than {@link Integer#MAX_VALUE}
-   * bytes.
+   * Validates user file buffer size configuration.
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/Configuration.java]: I think its valuable to document the valid values somewhere (maybe docs)

File: shell/src/main/java/alluxio/cli/ValidateConf.java [hunk 3]
```diff
@@ -31,15 +31,15 @@
...
-    if (Configuration.validate()) {
-      LOG.info("All configuration entries are valid.");
-    } else {
-      LOG.info("Configuration has invalid entries.");
-      ret = -1;
+    try {
+      Configuration.validate();
+      LOG.info("Configuration is valid.");
+    } catch (IllegalStateException e) {
+      LOG.warn("Configuration is invalid: {}", e.getMessage());
```
Comment (calvinjia) [inline shell/src/main/java/alluxio/cli/ValidateConf.java]: Log this at error level? I don't think we should put the stack trace, but the end result is crashing the system which seems to be more of an issue than warn.

File: core/common/src/main/java/alluxio/Configuration.java (hunk 1 above)
Comment (jsimsa) [inline core/common/src/main/java/alluxio/Configuration.java]: why not? either both should be set or neither, which is what I am going for here

File: shell/src/main/java/alluxio/cli/ValidateConf.java (hunk 3 above)
Comment (jsimsa) [inline shell/src/main/java/alluxio/cli/ValidateConf.java]: done

File: core/common/src/main/java/alluxio/Configuration.java (hunk 2 above)
Comment (jsimsa) [inline core/common/src/main/java/alluxio/Configuration.java]: done

File: core/common/src/main/java/alluxio/Configuration.java (hunk 1 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/Configuration.java]: I think it's reasonable for a user to set `zookeeper.enabled=false` but not set an address. Should this instead be `checkState(getBoolean(PropertyKey.ZOOKEEPER_ENABLED) == containsKey(PropertyKey.ZOOKEEPER_ADDRESS))`?

**Comment by jsimsa** [comment]
@calvinjia @calvinjia addressed your comments ... PTAL

File: core/common/src/main/java/alluxio/exception/PreconditionMessage.java [hunk 4]
```diff
@@ -55,8 +53,13 @@
   FILE_WRITE_LOCATION_POLICY_UNSPECIFIED("The location policy is not specified"),
   UFS_READ_LOCATION_POLICY_UNSPECIFIED("The UFS read location policy is not specified"),
   GCS_BUCKET_MUST_BE_SET("The %s system property must be set to use the GCSUnderStorageCluster"),
+  INCONSISTENT_ZK_CONFIGURATION(
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/exception/PreconditionMessage.java]: Update to reflect the previous change

Comment (jsimsa) [inline core/common/src/main/java/alluxio/exception/PreconditionMessage.java]: done

File: core/common/src/main/java/alluxio/Configuration.java [hunk 5]
```diff
@@ -493,58 +494,45 @@ private static void checkWorkerPorts() {
   }
 
   /**
-   * {@link PropertyKey#USER_FILE_BUFFER_BYTES} should not bigger than {@link Integer#MAX_VALUE}
-   * bytes.
+   * Validates the user file buffer size is a non-negative number.
    *
-   * @throws IllegalArgumentException if USER_FILE_BUFFER_BYTES bigger than Integer.MAX_VALUE
+   * @throws IllegalStateException if invalid user file buffer size configuration is encountered
```
Comment (apc999) [inline core/common/src/main/java/alluxio/Configuration.java]: should we just remove this ? we typically don't javadoc runtime error

Comment (aaudiber) [inline core/common/src/main/java/alluxio/Configuration.java]: We're moving in the direction of only using unchecked exceptions for issues indicating bugs, so a checked exception or return value would be preferable.

**Pull request merged**